test/compilable/test21196.d

```diff
@@ -0,0 +1,71 @@
+// https://issues.dlang.org/show_bug.cgi?id=21674
+// REQUIRED_ARGS: -de
+
+struct Module
+{
+    CachedString data;
+}
+
+struct CachedString
+{
+    private size_t len;
```

Comment:
Remaining question, if I were to added an extra field here...

Reply:
Yes, because without the `alias this`, compilation fails.

Reply:
So this should work:
```
struct CachedString
{
    void str(string value) { str = value.ptr; len = value.length; }
    string str() { return str[0..len]; }
    alias str this;
  public:
    int flag;
  private:
    const char* str;
    size_t len;
}
```
Because all we see is the CallExp, and we can't verify that the setter _really_ initializes the struct properly.